st2common/st2common/services/packs.py

```
@@ -215,3 +221,68 @@ def search_pack_index(
+        except PermissionError:
+            LOG.error(
+                'No permission to delete the "%s" file',
+                action_entrypoint_file_path,
+            )
+            msg = 'No permission to delete "{0}" file from disk'.format(
+                action_entrypoint_file_path
+            )
+            abort(
+                http_client.INTERNAL_SERVER_ERROR,
+                six.text_type(msg),
+            )
+        except Exception as e:
+            LOG.error(
+                'Unable to delete "%s" file. Exception was "%s"',
+                action_entrypoint_file_path,
+                e,
+            )
+            msg = (
+                'Delete operation unsuccessful. "{0}" file still exists on disk'.format(
+                    action_entrypoint_file_path
+                )
+            )
+            abort(http_client.INTERNAL_SERVER_ERROR, six.text_type(msg))
+
+    if os.path.exists(action_metadata_file_path):
+        try:
+            os.remove(action_metadata_file_path)
+        except PermissionError:
+            LOG.error(
+                'No permission to delete the "%s" file',
+                action_metadata_file_path,
+            )
+            msg = 'No permission to delete "{0}" file from disk'.format(
+                action_metadata_file_path
+            )
+            abort(
+                http_client.INTERNAL_SERVER_ERROR,
+                six.text_type(msg),
+            )
+        except Exception as e:
+            LOG.error(
+                'Unable to delete "%s" file. Exception was "%s"',
+                action_metadata_file_path,
+                e,
+            )
+            if os.path.isfile(action_metadata_file_path):
+                msg = 'Delete operation unsuccessful. "{0}" file still exists on disk'.format(
+                    action_metadata_file_path
+                )
+                abort(http_client.INTERNAL_SERVER_ERROR, six.text_type(msg))
```

Please throw an exception here instead of doing a http abort. Let the API controller do the abort for the http request on exception.